adapters/repos/db/aggregator/filtered.go

```diff
@@ -38,20 +39,30 @@ func (fa *filteredAggregator) Do(ctx context.Context) (*aggregation.Result, erro
 	// without grouping there is always exactly one group
 	out.Groups = make([]aggregation.Group, 1)
 
+	filter := fa.getFilterOrDefault(fa.params.Filters)
 	s := fa.getSchema.GetSchemaSkipAuth()
 	ids, err := inverted.NewSearcher(fa.store, s, fa.invertedRowCache, nil,
 		fa.Aggregator.classSearcher, fa.deletedDocIDs, fa.stopwords, fa.shardVersion).
-		DocIDs(ctx, fa.params.Filters, additional.Properties{},
+		DocIDs(ctx, filter, additional.Properties{},
 			fa.params.ClassName)
 	if err != nil {
 		return nil, errors.Wrap(err, "retrieve doc IDs from searcher")
 	}
 
+	var idsList []uint64
+	if len(fa.params.SearchVector) > 0 {
+		idsList, err = fa.searchByVector(fa.params.SearchVector, fa.params.Limit, ids)
+		if err != nil {
+			return nil, err
+		}
+	} else {
+		idsList = flattenAllowList(ids)
+	}
+
```

Comment:
I think that also here you need to take into account the `limit` param that user can pass in the `Aggregate` query and make the results cut